src/com/walmartlabs/lacinia/pedestal.clj

```diff
@@ -359,22 +362,37 @@
                 (resolve/on-deliver! resolver-result
                                      (fn [result]
                                        (deliver *result result)))
-                (apply-result-to-context context @*result)))}))
+                (let [result @*result]
+                  ;; Lacinia changed the contract here is 0.36.0 (to support timeouts), the result
+                  ;; maybe an exception thrown during initial processing of the query.
+                  (if (instance? Throwable result)
+                    (do (log/error :event :execution-exception
+                                   :ex result)
+                        ;; Raise the exception again for another interceptor to handle.
+                        ;; If unhandled, eventually ends up in error-response-interceptor
+                        (throw result))
+                    (apply-result-to-context context result)))))}))
 
 (def ^{:added "0.2.0"} async-query-executor-handler
   "Async variant of [[query-executor-handler]] which returns a channel that conveys the
   updated context."
   (interceptor
     {:name ::async-query-executor
      :enter (fn [context]
-              (let [ch (chan 1)
-                    resolver-result (execute-query context)]
+              (let [resolver-result (execute-query context)
+                    *result         (promise)]
                 (resolve/on-deliver! resolver-result
                                      (fn [result]
-                                       (->> result
-                                            (apply-result-to-context context)
-                                            (put! ch))))
-                ch))}))
+                                       (deliver *result result)))
+                (go (let [result @*result]
+                      ;; Lacinia changed the contract here is 0.36.0 (to support timeouts), the result
+                      ;; maybe an exception thrown during initial processing of the query.
+                      (if (instance? Throwable result)
+                        (do (log/error :event :execution-exception
+                                       :ex result)
+                            ;; TODO: Async exceptions should also go into the error interceptor
```

Comment:
This is the TODO that should be resolved.

Reply:
When thats done, probably also remove the `go` in favour of a normal `chan`. If you have suggestions on the pedestal async stuff, let me know, otherwise I'll invest some time (probably) over the weekend.

Reply:
Hey there. This has been resolved.

After talking to the Pedestal people, it is indeed correct that [this function `ex-info->throwable`](https://github.com/pedestal/pedestal/blob/4edaa5c0b4ec33a9ba356fadda3a9651dc050b5b/interceptor/src/io/pedestal/interceptor/chain.clj#L32-L42) is private, but you can assemble the same information from the context map + which interceptor will assoc the error in the map. Its sad that we have to duplicate a bit of pedestal functionality, but we do now have the same error handling on the sync- and async paths.